app/src/main/java/com/hedvig/app/feature/referrals/ReferralsViewModel.kt

```diff
@@ -0,0 +1,26 @@
+package com.hedvig.app.feature.referrals
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hedvig.android.owldroid.graphql.ReferralsQuery
+import kotlinx.coroutines.launch
+
+abstract class ReferralsViewModel : ViewModel() {
+    abstract val data: LiveData<ReferralsQuery.Data>
+}
+
+class ReferralsViewModelImpl(
+    private val referralsRepository: ReferralsRepository
+) : ReferralsViewModel() {
+    override val data = MutableLiveData<ReferralsQuery.Data>()
+
+    init {
+        viewModelScope.launch {
+            val result = runCatching { referralsRepository.referralsAsync().await() }
+            // TODO: Error state
```

Comment:
Act on it?

Reply:
I'm still waiting for design for this state

Reply:
I'm deferring this implementation to a later PR.